oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/spi/editor/FulltextDocumentMaker.java

```diff
@@ -336,6 +339,37 @@ private boolean indexProperty(String path,
         return dirty;
     }
 
+    static String getValueAsTruncatedString(Object val) {
+        String value = val.toString();
+        boolean isArray = value.startsWith("[") && value.endsWith("]");
+        return value.length() <= LOG_TRUNCATION_LENGTH ? value
+            : value.substring(0, LOG_TRUNCATION_LENGTH) + "..." + (isArray ? "]" : "");
+    }
+
+    static String truncateForLogging(PropertyState property) {
+        Type<?> t = property.getType();
+        if (t.isArray()) {
+            if (property.count() == 0) {
+                return "[]";
+            }
+
+            Type<?> base = t.getBaseType();
+            StringBuilder sb = new StringBuilder();
+            sb.append("[");
+            for (int i = 0; i < property.count(); i++) {
+                sb.append(property.getValue(base, i));
+                if (i < property.count() - 1) {
+                    sb.append(", ");
+                }
+            }
+            sb.append("]");
```

Comment:
This would be cleaner in functional style, something like:

```
IntStream.range(0, property.count())
        .mapToObj(i -> property.getValue(Type.STRING, i))
        .collect(Collectors.joining(",", "[", "]"));
```
Maybe less efficient, but this is code is not in a hotspot, so it does not matter.

Reply:
Yeah I like this better